shared/src/business/entities/DocketEntry.ts

```diff
@@ -0,0 +1,592 @@
+import { WorkItem } from './WorkItem';
+import {
+  createISODateAtStartOfDayEST,
+  createISODateString,
+} from '../utilities/DateHandler';
+
+export class DocketEntryClass {
+  public entityName: string;
+  public action: string;
+  public additionalInfo: string;
+  public additionalInfo2: string;
+  public addToCoversheet: boolean;
+  public archived: boolean;
+  public attachments: string;
+  public certificateOfService: string;
+  public certificateOfServiceDate: string;
+  public createdAt: string;
+  public date: string;
+  public docketEntryId: string;
+  public docketNumber: string;
+  public docketNumbers: string;
+  public documentContentsId: string;
+  public documentIdBeforeSignature: string;
+  public documentTitle: string;
+  public documentType: string;
+  public eventCode: string;
+  public filedBy: string;
+  public filingDate: string;
+  public freeText: string;
+  public freeText2: string;
+  public hasOtherFilingParty: string;
+  public hasSupportingDocuments: string;
+  public index: string;
+  public isAutoGenerated: string;
+  public isFileAttached: string;
+  public isLegacy: string;
+  public editState: string;
+  public isLegacySealed: boolean;
+  public isLegacyServed: boolean;
+  public isMinuteEntry: boolean;
+  public isOnDocketRecord: boolean;
+  public isPaper: boolean;
+  public isPendingService: boolean;
+  public isSealed: boolean;
+  public isStricken: boolean;
+  public lodged: string;
+  public mailingDate: string;
+  public numberOfPages: string;
+  public objections: string;
+  public sealedTo: string;
+  public filers: string[];
+  public ordinalValue: string;
+  public otherFilingParty: string;
+  public partyIrsPractitioner: string;
+  public processingStatus: string;
+  public receivedAt: string;
+  public relationship: string;
+  public scenario: string;
+  public secondaryDocument: string;
+  public servedAt: string;
+  public servedPartiesCode: string;
+  public serviceDate: string;
+  public serviceStamp: string;
+  public strickenAt: string;
+  public supportingDocument: string;
+  public trialLocation: string;
+  public userId: string;
+  public privatePractitioners: any[];
+  public servedParties: any[];
+  public signedAt: string;
+  public draftOrderState: string;
+  public stampData: string;
+  public isDraft: string;
+  public judge: string;
+  public judgeUserId: string;
+  public pending: string;
+  public previousDocument: {
+    docketEntryId: string;
+    documentTitle: string;
+    documentType: string;
+  };
+  public qcAt: string;
+  public qcByUserId: string;
+  public signedByUserId: string;
+  public signedJudgeName: string;
+  public signedJudgeUserId: string;
+  public strickenBy: string;
+  public strickenByUserId: string;
+  public workItem: any;
+
+  constructor(
+    rawDocketEntry: any,
+    {
+      applicationContext,
+      petitioners,
+    }: { applicationContext: IApplicationContext; petitioners?: any[] },
+  ) {
+    this.entityName = 'DocketEntry';
```

Comment:
NitPick: I might set entityName on init rather than the constructor so that the constructor is doing less. 
So -> public EntityName: string = 'DocketEntry'